src/pages/OfferPage.js

```diff
@@ -0,0 +1,168 @@
+import React, { useEffect, useContext, useState, useCallback } from "react";
+import { useParams, Redirect } from "react-router-dom";
+
+import OfferWidget from "../components/HomePage/SearchResultsArea/Offer/OfferWidget";
+import Offer from "../components/HomePage/SearchResultsArea/Offer/Offer";
+import { CardContent } from "@material-ui/core";
+
+import {
+    disableOffer as disableOfferService,
+    hideOffer as hideOfferService,
+    enableOffer as enableOfferService,
+} from "../services/offerVisibilityService";
+import useOffer from "../hooks/useOffer";
+
+export const OfferPageControllerContext = React.createContext();
+
+export const OfferPageController = () => {
+    const { id } = useParams();
+    const [offer, setOffer] = useState(null);
+    const [loading, setLoading] = useState(true);
```

Comment:
You can remove these state variables. The `offer` variable will be the `offerData`, and the loading you can return from the `useOffer` hook (similar to `isValidating`  in the `useSession` hook)

Then you won't need the useEffect below either to sync the variables

Reply:
The first time `useSwr` is called, `isValidating` is false. I'll be using `!data` to set `loading` instead.

![Screenshot from 2021-09-02 08-13-36](https://user-images.githubusercontent.com/50210213/131799650-26582e92-70d8-4389-81da-2026af505834.png)
![Screenshot from 2021-09-02 08-13-12](https://user-images.githubusercontent.com/50210213/131799679-bb75ebbb-3fb0-465c-b9a3-43d1e48403d0.png)

Reply:
I thought isValidating was !data 😅 you can use it that way sure

Reply:
To be clear, you can export the loading variable from the hook, setting it to !data there (not on the component)